#!/usr/bin/env Rscript
library("optparse")
 
option_list <- list(
    make_option(c("-p", "--peaks_sites_file"), type="character", help="a file containing peak coordinates generated by FindPeaks"),
    make_option(c("-g", "--gtf_file"), type="character", help="reference (GTF) file"),
    make_option(c("-b", "--bamfile"), type="character", help="scRNA-seq BAM file"),
    make_option(c("-w", "--whitelist_file"), type="character", help="file of cell barcodes to count"),
    make_option(c("-o", "--output_dir"), type = "character", default = "sierra_out", help = "name of directory to write output (will be created if it doesn't exist) [default = %default]"),
    make_option(c("-c", "--ncores"), type = "integer", default = 1, help = "Number of cores[default= %default]"),
    make_option(c("-s", "--gene_symbol_ref"), type = "character", default = "gene_name", help = "field in the GTF file containing the gene symbol"),
    make_option(c("-u", "--noUMI"), action="store_true", default=FALSE, help = "Boolean. Do not count UMI [default= %default]")
)
 
opt_parser <- OptionParser(option_list=option_list)
opt <- parse_args(opt_parser)

# print(opt)


library(Sierra)
CountPeaks(peak.sites.file = opt$peaks_sites_file, 
           gtf.file = opt$gtf_file,
           bamfile = opt$bamfile, 
           whitelist.file = opt$whitelist_file,
           output.dir = opt$output_dir, 
           ncores = opt$ncores,
           gene.symbol.ref = opt$gene_symbol_ref,
           countUMI = !opt$noUMI)
